client/src/components/Workflow/Editor/Forms/FormTool.vue

```diff
@@ -1,32 +1,37 @@
 <template>
     <ToolCard
-        v-if="hasData"
-        :id="configForm.id"
-        :version="configForm.version"
-        :title="configForm.name"
-        :description="configForm.description"
+        :id="configForm?.id || ''"
+        :version="configForm?.version"
+        :title="configForm?.name || step.name"
+        :description="configForm?.description"
         :options="configForm"
         :message-text="messageText"
         :message-variant="messageVariant"
+        :step-error="step.errors"
         @onChangeVersion="onChangeVersion"
         @onUpdateFavorites="onUpdateFavorites">
         <template v-slot:body>
             <FormElement
+                v-if="hasData"
```

Comment:
label and annotation should both remain visible and editable, I think

Reply:
It shows nothing when the tool is not installed. Only the headers are visible.

Reply:
Are you describing what should happen or what currently happens ? annotation and label are not properties of the tool, so they can be shown.

Reply:
Gotcha! I just checked again. Label and annotation can be edited but a workflow with a missing tool can not be saved:
<img width="765" alt="image" src="https://github.com/galaxyproject/galaxy/assets/8046843/7f613363-5718-4aee-9bbe-152511a34e43">

If this is correct, then I think would be better to disable editing and inform the user that these issues need to be resolved before they can save and continue editing. Otherwise, yes, these conditions should be dropped.

Reply:
Yeah, can't be saved, but I don't think the annotation field needs to concern itself with that. You can for instance move steps around and not save them until you've replaced a tool. That said if you want to disable (i.e. make the field non-editable) that is fine too.

Reply:
OK, thanks! I will make both fields non-editable.

By the way, when I try to make the fields non-editable, the `FormElement` input completely [disappears](https://github.com/galaxyproject/galaxy/blob/dev/client/src/components/Form/FormElement.vue#L224) because [`showField`](https://github.com/galaxyproject/galaxy/blob/dev/client/src/components/Form/FormElement.vue#L129) becomes false. Can you explain why this happens? Also, I noticed that there’s no place where `disabled` is assigned for `FormElement`. I’m not sure about the reason for the existence of `showField`.